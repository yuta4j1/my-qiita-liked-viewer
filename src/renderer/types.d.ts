export type ArticleInfo = {
  uuid: string;
  title: string;
  url: string;
  userImageUrl: string;
  updateDate: string;
  tagList: string[];
};

export type Column = {
  categoryName: string;
  articles: ArticleInfo[];
};

export type SourceArticleInfo = {
  uuid: string;
  title: string;
  url: string;
  userImageUrl: string;
  updateDate: string;
  tagList: string[];
};

export type ModalState = {
  isOpen: boolean;
  headerTitle?: string;
  modalBody: JSX.Element;
  atClose: (any) => void;
};

export type SideModalState = {
  isShow: boolean;
};

export type ColumnState = string[];

export type ColumnPosition = number;

export type LoadingState = {
  isActive: boolean;
};

export type ColumnCardState = {
  id: string;
  text: string;
  order: number;
};

export type User = {
  id: string;
  name: string;
  profileImageUrl: string;
};

export type ReduxAction<T> = {
  type: string;
  payload: T;
};
