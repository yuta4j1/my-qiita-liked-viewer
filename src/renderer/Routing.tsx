import * as React from 'react'
import {
  Route,
  Switch,
  BrowserRouter as Router,
  withRouter
} from 'react-router-dom'
import App from '@/components/App'
import LoginWrapper from '@/components/Login/LoginWrapper'

const Routing = (
  <Router>
    <Switch>
      <Route path="/view" component={App} />
      <Route path="/" component={withRouter(LoginWrapper)} />
    </Switch>
  </Router>
)

export default Routing
