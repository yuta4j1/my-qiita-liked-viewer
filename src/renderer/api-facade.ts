import axiosBase, { AxiosPromise } from 'axios';
import { resolve } from 'path';

const requestTo = 'http://localhost:8080/mql';
const defaultConfig = {
    timeout: 20000,
    headers: {
        "Content-Type": "application/json"
    },
    responseType: 'json'
}
const axios = axiosBase.create({
    baseURL: requestTo,
    ...defaultConfig
});
export function get(url: string): AxiosPromise<any> | void {
    axios.get(url).then((response) => {
        console.log(response);
        resolve(JSON.stringify(response));
    }).then((error) => {
        console.log(error);
    });

}