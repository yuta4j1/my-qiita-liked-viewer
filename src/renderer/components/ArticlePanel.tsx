import * as React from 'react';
import styled from '../theme/index';
import { ArticleInfo } from '../types';

type ArticleProps = ArticleInfo;
type ArticleState = ArticleInfo;

const Panel = styled.div`
  display: flex;
  align-items: center;
  border-style: solid none solid none;
  border-width: 0.25px;
  border-color: #eeeeee;
`;

const UploadDate = styled.div`
  font-size: 10.9px;
`;

const UserIcon = styled.div`
  padding: 20px;
  align-items: center;
  border-style: solid;
  border-width: 0.5px;
  border-radius: 50%;
  border-color: #bdbdbd;
`;

const Item = styled.div`
  margin: 10px;
`;

const TagList = styled.ul`
  display: flex;
  list-style: none;
`;

const ATag = styled.li`
  margin: 0px 5px;
`;

class ArticlePanel extends React.Component<ArticleProps, ArticleState> {
  constructor(props: ArticleProps) {
    super(props);
    this.state = { ...props };
  }

  createTagBadges() {
    return this.state.tagList.map((tagKey, tag) => {
      return <ATag key={tagKey}>{tag}</ATag>;
    });
  }

  render() {
    return (
      <Panel>
        <Item>
          <UserIcon>User</UserIcon>
        </Item>
        <Item>
          <UploadDate>
            <p>投稿日付：2019/12/31</p>/** TODO this.state.UploadDate */
          </UploadDate>
          <div>
            <a href="#">{this.state.title}</a>
          </div>
          <div>
            <TagList>{this.createTagBadges()}</TagList>
          </div>
        </Item>
      </Panel>
    );
  }
}

export default ArticlePanel;
