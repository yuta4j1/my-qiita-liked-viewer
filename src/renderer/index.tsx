import { render } from 'react-dom'
import routing from './Routing'

render(routing, document.getElementById('app'))
