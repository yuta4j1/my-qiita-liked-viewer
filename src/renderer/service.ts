import { shell } from 'electron'

export function openExternalBrowser(url: string): void {
  shell.openExternal(url)
}
