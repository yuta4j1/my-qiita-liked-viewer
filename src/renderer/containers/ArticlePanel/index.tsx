import * as React from 'react'
import { Dispatch } from 'redux'
import { useSelector, useDispatch } from 'react-redux'
import { ArticleInfo, ColumnState } from '@/types'
import { AppState } from '../store'
import { settingColumn } from '../ColumnSetting/action'
import ArticlePanel from '@/components/ArticlePanel'

type Props = ArticleInfo

const ArticlePanelContainer: React.FC<Props> = props => {
  const columns = useSelector((state: AppState) => state.columnStateReducer)

  const dispatch = useDispatch<Dispatch>()

  const setColumns = React.useCallback(
    (data: ColumnState) => {
      dispatch(settingColumn(data))
    },
    [dispatch]
  )

  const childProps = { ...props, columns, setColumns }

  return <ArticlePanel {...childProps} />
}

export default ArticlePanelContainer
