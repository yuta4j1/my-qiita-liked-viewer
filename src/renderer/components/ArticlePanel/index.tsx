import * as React from 'react';
import { useState } from 'react';
import { openExternalBrowser } from '@/service';
import styled from '@/theme';
import { ArticleInfo, ColumnState } from '@/types';

type ArticleProps = ArticleInfo & {
  columns: ColumnState;
  setColumns: (data: ColumnState) => void;
};

const Panel = styled.div`
  display: flex;
  align-items: center;
  border-style: solid none solid none;
  border-width: 0.25px;
  border-color: #eeeeee;
`;

const UploadDate = styled.div`
  font-size: 10.9px;
`;

const Item = styled.div`
  margin: 10px;
`;

const TagList = styled.div`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  letft: 20%;
`;

const ATag = styled.div`
  margin: 2px 2px;
  padding: 5px;
  border-radius: 10px;
  background-color: #f5f5f5;
  font-family: 'Assistant', sans-serif;
  font-size: 1em;
  cursor: pointer;
  transition: color .3s
  &:hover {
    color: #2196f3;
  }
`;

const Link = styled.a`
  font-family: 'Noto Sans JP', sans-serif;
  text-decoration: none;
  color: #337ab7;
`;

const UserImage = styled.img`
  border-radius: 50%;
  width: 100px;
  height: 100px;
  background: url(${props => props.src});
`;

const ArticlePanel: React.FC<ArticleProps> = props => {
  const [state] = useState({ ...props });

  const tagClickEvent = (tag: string) => {
    props.setColumns([...props.columns, tag]);
  };

  const createTagBadges = () =>
    state.tagList.map((tag, key) => {
      return (
        <ATag key={key} onClick={e => tagClickEvent(tag)}>
          {tag}
        </ATag>
      );
    });

  const openArticle = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
    e.preventDefault();
    openExternalBrowser('https://qiita.com/' + state.url);
  };

  return (
    <Panel>
      <Item>
        <div>
          <UserImage src={state.userImageUrl} />
        </div>
      </Item>
      <Item>
        <UploadDate>
          <p>投稿日付：{state.updateDate}</p>
        </UploadDate>
        <div>
          <Link href="#" onClick={e => openArticle(e)}>
            {state.title}
          </Link>
        </div>
        <div>
          <TagList>{createTagBadges()}</TagList>
        </div>
      </Item>
    </Panel>
  );
};

export default ArticlePanel;
