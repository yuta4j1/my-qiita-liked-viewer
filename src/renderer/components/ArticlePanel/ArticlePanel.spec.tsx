import * as React from 'react';
import renderer from 'react-test-renderer';
import ArticlePanel from './index';
import { ArticleInfo } from '../../types';

describe('ArticlePanel test', () => {
  it('<ArticlePanel /> snapshot', () => {
    const props: ArticleInfo[] = [
      {
        tagList: ['Git'],
        title:
          'Git で force pull する方法 ( $ git reset --hard origin/branch_name )',
        updateDate: '2015/11/04',
        url: '/Yinaura/items/30992f25b87f76af3ba2',
        userImageUrl:
          'https://qiita-image-store.s3.amazonaws.com/0/90607/profile-images/1473705145',
        uuid: '30992f25b87f76af3ba2'
      },
      {
        tagList: ['Go'],
        title: 'Goでパスワード入力を読み取る(Windows/Mac/Linux対応)',
        updateDate: '2016/06/27',
        url: '/moutend/items/12d53750363edbbc3d6b',
        userImageUrl:
          'https://qiita-image-store.s3.amazonaws.com/0/89489/profile-images/1473704769',
        uuid: '12d53750363edbbc3d6b'
      }
    ];

    const tree = renderer
      .create(props.map((v, i) => <ArticlePanel key={i} {...v} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
