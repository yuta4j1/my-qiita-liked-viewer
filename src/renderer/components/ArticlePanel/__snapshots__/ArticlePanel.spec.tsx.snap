// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePanel test <ArticlePanel /> snapshot 1`] = `
Array [
  <div
    className="sc-bdVaJa jWHsEd"
  >
    <div
      className="sc-htpNat CURwj"
    >
      <div>
        <img
          height={100}
          src="https://qiita-image-store.s3.amazonaws.com/0/90607/profile-images/1473705145"
          width={100}
        />
      </div>
    </div>
    <div
      className="sc-htpNat CURwj"
    >
      <div
        className="sc-bwzfXH gkBGFQ"
      >
        <p>
          投稿日付：
          2015/11/04
        </p>
      </div>
      <div>
        <a
          className="sc-EHOje kWEsDe"
          href="#"
          onClick={[Function]}
        >
          Git で force pull する方法 ( $ git reset --hard origin/branch_name )
        </a>
      </div>
      <div>
        <div
          className="sc-bxivhb hXdqup"
        >
          <div
            className="sc-ifAKCX eHpeHy"
          >
            Git
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-bdVaJa jWHsEd"
  >
    <div
      className="sc-htpNat CURwj"
    >
      <div>
        <img
          height={100}
          src="https://qiita-image-store.s3.amazonaws.com/0/89489/profile-images/1473704769"
          width={100}
        />
      </div>
    </div>
    <div
      className="sc-htpNat CURwj"
    >
      <div
        className="sc-bwzfXH gkBGFQ"
      >
        <p>
          投稿日付：
          2016/06/27
        </p>
      </div>
      <div>
        <a
          className="sc-EHOje kWEsDe"
          href="#"
          onClick={[Function]}
        >
          Goでパスワード入力を読み取る(Windows/Mac/Linux対応)
        </a>
      </div>
      <div>
        <div
          className="sc-bxivhb hXdqup"
        >
          <div
            className="sc-ifAKCX eHpeHy"
          >
            Go
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
